R/get_lifetable.R

```diff
@@ -0,0 +1,140 @@
+#' Get life table data
+#' 
+#' @description
+#' This function allows you to download the life table data from the 
+#' UN Population division, available here (https://population.un.org/wpp/Download/Files/1_Indicators%20(Standard)/CSV_FILES/). 
+#' The data is downloaded to a temporary file and returned as a 
+#' dataframe. Any new download will override the old one. You can specify country,
+#' gender, time span, and span of age groups. Currently all available life table 
+#' data is returned. For death rates specifically, see get_deathrates().
+#' 
+#' @details
+#' Here are the terms of use for all UN produced data: https://www.un.org/en/about-us/terms-of-use.
+#' Please look through before downloading the data. The data can be copied, 
+#' altered, and distributed, even for commercial use.
+#' 
+#' One issue with this function is that the complete csv is 140 MB. 
+#' Currently the usage of the vroom package means that memory usage is minimized, 
+#' as the vroom function downloads pointers and only loads what is actually called.
+#' However, an initial check of available memory could be added in order to be prudent.
+#' 
+#' Future additions:
+#' Memory check. Check of years and ages inputs.
+#' Spell checks. Url check - make sure its still there.
+#' 
+#' @param country all, iso3c code, or country name in English. If not available or 
+#' misspelled, will throw error. To do: add a list of all countries/regions that we 
+#' have data for.
+#' @param gender both (only total), female, male, all (female, male, and total)
+#' @param years.start Life table data is available for a range of years. This is 
+#' the first year from which life table data will be returned. Minimum is 1950.
+#' @param years.end The end of the period of life table data that is returned. Maximum is 2100. 
+#' @param ages.start Life table data starts at age 0 (default), and is bounded in 5 year age
+#' groups.If this age is not available, will return 
+#' a table from the lowest bound of the age group older than the age specified.
+#' @param ages.end This is the last age of the specified life table. Since the life table is in groups, will 
+#' return a life table where the oldest age group includes the ages.end value. 
+#'
+#' @keywords UN, demography, download, population, life table, deathrates, mx, ax, dx, gender, age groups
+#' @return Data frame of life table data.
+#' @export
+#' 
+#' @import vroom
+#' @import countrycode
+#' @import tidyr
+#' @import data.table
+#' @import stringr
+#' 
+#' @examples
+#' get_lifetable(country="NZA")
+get_lifetable <- function(country="all", 
```

Comment:
I see this output has `AgeGrp AgeGrpStart AgeGrpSpan` columns - could we maybe have a AgeGrpEnd column too?